src/main/java/tfifteenfour/clipboard/logic/parser/AddCommandParser.java

```diff
@@ -34,26 +39,72 @@ public class AddCommandParser implements Parser<AddCommand> {
+        case ADD_MODULE:
+            Course course = parseCourseInfo(args);
+            return new AddCourseCommand(course);
+        case ADD_GROUP:
+            Group group = parseGroupInfo(args);
+            return new AddGroupCommand(group);
+        case ADD_SESSION:
+            return new AddSessionCommand();
+        case ADD_STUDENT:
+            Student student = parseStudentInfo(args);
+            return new AddStudentCommand(student);
+        default:
+            throw new ParseException("Invalid argument for add command");
+        }
+    }
+
+
+    private Course parseCourseInfo(String args) throws ParseException {
+        String[] tokens = ArgumentTokenizer.tokenizeString(args);
+        if (tokens.length > 3) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCourseCommand.MESSAGE_USAGE));
+        }
+
+        Course course = ParserUtil.parseCourse(tokens[2]);
+        return course;
+    }
+
+    private Group parseGroupInfo(String args) throws ParseException {
+        String[] tokens = ArgumentTokenizer.tokenizeString(args);
+        if (tokens.length > 3) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddGroupCommand.MESSAGE_USAGE));
+        }
+
+        Group group = ParserUtil.parseGroup(tokens[2]);
+        return group;
+    }
+
+    private Student parseStudentInfo(String args) throws ParseException {
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_STUDENTID,
-                        PREFIX_MODULE, PREFIX_REMARK, PREFIX_TAG);
+                ArgumentTokenizer.tokenizePrefixes(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_STUDENTID, PREFIX_REMARK, PREFIX_TAG);
 
-        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_STUDENTID, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_MODULE)
-                || !argMultimap.getPreamble().isEmpty()) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_STUDENTID, PREFIX_PHONE, PREFIX_EMAIL)
+                || !AddCommandType.isValidAddType(argMultimap.getPreamble())) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddStudentCommand.MESSAGE_USAGE));
         }
 
         Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
         Phone phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
         Email email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
         StudentId studentId = ParserUtil.parseStudentId(argMultimap.getValue(PREFIX_STUDENTID).get());
-        Set<Course> modules = ParserUtil.parseModules(argMultimap.getAllValues(PREFIX_MODULE));
+        Set<Course> modules = ParserUtil.parseModules(argMultimap.getAllValues(PREFIX_COURSE));
```

Comment:
```suggestion
        Set<Course> courses = ParserUtil.parseCourses(argMultimap.getAllValues(PREFIX_COURSE));
```

Reply:
Commented on this as other instances in `ParserUtil` like `parseCourse` has already been changed, so we can just change this for consistency first. Other instances of `modules` we can change when we get together as a team?

Reply:
Changes made, but commented out anyway as Student should no longer reference modules and tags